packages/material-ui/src/ListItem/ListItem.js

```diff
@@ -99,7 +99,7 @@ const ListItem = React.forwardRef(function ListItem(props, ref) {
     component: componentProp,
     ContainerComponent = 'li',
     ContainerProps: { className: ContainerClassName, ...ContainerProps } = {},
-    dense,
+    dense = false,
```

Comment:
The state can come from the parent list too, it would override it.

```suggestion
    dense,
```

Reply:
How would a default value override it?

Reply:
Oh ok, my bad, it won't. I have assumed we were using `dense !== undefined`.

But shouldn't we change the merge dense check logic? If the parent can be not dense but child dense, why prevent the opposite (parent dense but child not dense)?